packages/components/src/styles/components/app-header.scss

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// APP-HEADER
+//
+
+@use "../mixins/interactive-dark-theme" as *;
+
+// ----------------
+
+// TODO: Create app-header tokens:
+:root {
+  --token-app-header-height: 60px;
+  --token-app-header-home-link-padding: 4px;
+}
+
+.hds-app-header {
+  position: fixed;
+  top: 0;
+  right: 0;
+  left: 0;
+  z-index: 20; // needs to stay above the main content
+  display: flex;
+  gap: 12px;
+  align-items: center;
+  justify-content: space-between;
+  min-height: var(--token-app-header-height);
+  padding: 12px 16px 11px 16px; // bottom padding of 11px is to account for the bottom border
+  color: var(--token-color-foreground-high-contrast);
+  background: var(--token-color-palette-neutral-700);
+  border-bottom: 1px solid var(--token-color-palette-neutral-500);
+
+  // RESPONSIVE VIEWS
+
+  // Mobile/small and tablet/medium compound view: applies at tablet min-width and below
+  &.hds-app-header--is-mobile {
+    &:not(.hds-app-header--menu-is-open) {
+      .hds-app-header__menu {
+        display: none; // dropdown menu is hidden when isOpen = false
+      }
+    }
+
+    // Style content as dropdown menu in mobile/tablet view
+    .hds-app-header__menu {
+      position: absolute;
+      top: calc(100% + 1px); // We add 1px to account for the AppHeader bottom border
+      right: 0;
+      left: 0;
+      flex-wrap: wrap;
+      align-content: flex-start;
+      height: calc(100vh - var(--token-app-header-height));
+      padding: 16px;
+      overflow: auto; // allow users to scroll within the menu if the content is too long
+      background: var(--token-color-palette-neutral-700);
+    }
+
+    // Force nested dropdown content to align left in mobile/tablet view
+    .hds-dropdown__content {
+      left: 0;
+    }
```

Comment:
I can either delete the above styles forcing dropdown list position (and await collision detection being added to `Dropdown`) or else add more to better control the list position within all responsive views (which could always be deleted later when no longer needed.)

Reply:
UPDATE: I deleted styles forcing Dropdown list positioning in anticipation of @alex-ju 's updates to the Dropdown.